### PR TITLE
prevent warning message

### PR DESCRIPTION
**Pull request by Rom1-B**
Prevent:

image

```
PHP Warning (2): Undefined array key "id" in .../plugins/glpiinventory/inc/taskjobstate.class.php at line 626
```

### CODE REVIEW

File: inc/taskjobstate.class.php [hunk 1]
```diff
@@ -623,7 +623,7 @@ public function showStatesForComputer($computers_id)
 
        // Get the agent of the computer
         $agent->getFromDBByCrit(['itemtype' => 'Computer', 'items_id' => $computers_id]);
-        $agents_id = $agent->fields['id'];
+        $agents_id = $agent->fields['id'] ?? 0;
```
Comment (cedric-anne) [inline inc/taskjobstate.class.php:626]:
```suggestion
        $agents_id = $agent->fields['id'];
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]
Please add an entry in the changelog, with a functional description. It guess this issue happens when the `Tasks / Groups` tab is displayed on a computer linked to an invalid agent.

**Comment by Rom1-B** [comment]
> Please add an entry in the changelog, with a functional description. It guess this issue happens when the `Tasks / Groups` tab is displayed on a computer linked to an invalid agent.

In my case, the computer never had an agent, it was created manually.

This fix doesn't really have any functional impact, it just avoids a warning

But it does raise a question: does the tab make sense if the computer doesn't have an agent? In that case, should it only be displayed if there's a linked agent?

File: inc/taskjobstate.class.php [hunk 2]
```diff
@@ -118,7 +118,13 @@ public function getTabNameForItem(CommonGLPI $item, $withtemplate = 0)
     {
         switch ($item->getType()) {
             case 'Computer':
-                return __("Tasks / Groups", "glpiinventory");
+                if (
+                    method_exists($item, 'getInventoryAgent')
```
Comment (trasher) [inline inc/taskjobstate.class.php:122]: Method must exist on `Computer`, it seems useless to check it

Comment (Rom1-B) [reply inc/taskjobstate.class.php:122]: Yes, but without this check the CI is not happy:

https://github.com/glpi-project/glpi-inventory-plugin/pull/489/commits/e0567abd3e45d4a46a66de16d08aaf43e4e638c4

Comment (trasher) [reply inc/taskjobstate.class.php:122]: it can be solved using `/** @var Computer $item` I guess

Comment (trasher) [reply inc/taskjobstate.class.php:122]: (not a blocker anyway)

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**